packages/common/src/types/metricQuery.ts

```diff
@@ -36,3 +36,20 @@ export type SortField = {
     fieldId: string; // Field must exist in the explore
     descending: boolean; // Direction of the sort
 };
+
+export const idPattern = /((.+)(?<!val)id$)/i;
+export const extractEntityNameFromIdColumn = (
```

Comment:
I'd love to see a unit test for this function

Reply:
They already exist in `packages/backend/src/dbt/translator.test.ts` I just kept them there since we haven't setup common package to run tests yet.